### DISCUSSION TITLE
作者大大能否在获取云主机的时候支持切换到公网ip，以及资源告警时加上公网ip

### DISCUSSION THREAD

**Discussion by eavin7456** [starsliao/ConsulManager#33] [Ideas]
1.现在获取到的只能是内网IP，虽然功能上支持自定义IP，但问题是当有大批量的云主机的时这个步骤就会极为头疼，如果能在获取的时候支持切换到公网ip 那这样的话就方便很大

2.前段时间云主机到期告警的功能终于上线了。但是一直有一个问题，通过钉钉告警的资源内容里没有IP这个参数
这样的话就单看告警没法第一时间知道是哪个机器的IP,因为能看到这个告警的人不一定能登录阿里云，所以他们这种时候只知道有机器到期了，却不知道是哪一台

**Comment by starsliao** [comment]
1.目前已支持了各云设置优先获取ECS公网IP。
2.阿里云的资源到期接口没有提供IP信息，所以到期信息通知这块没有输出IP信息来，后续我增加上。
